go-controller/pkg/node/ovspinning/ovspinning_linux.go

```diff
@@ -18,7 +18,7 @@ import (
 )
 
 // These variables are meant to be used in unit tests
-var tickDuration time.Duration = 10 * time.Second
+var tickDuration time.Duration = 1 * time.Second
```

Comment:
doesn't this mean we are going to be checking more often than before? does that put load on the goroutine to do this more frequently? With the new IC model we want to cut down as much as possible on each node pod. I hope this has no scale implications... Kubelet's CPU manager itself is configured for 5 seconds then I vote 5 as a compromise.. but I will let @zeeke or @trozet chime in because i don't have data backing my comment here so I could be wrong.

Reply:
The issue with 5 as a compromise is that this makes it 10 in total. Because this needs to wait for the cpu manager to update the affinity of ovnkube-node, then notice that change and apply it. CPU manager goes over all containers in the system. This just reads the affinity most of the time.

Reply:
And just to answer the question explicitly - reading the affinity of a process should be really cheap and that is what is done every second. The code only applies the new affinity and causes thread rebalacing in OVS when a new guaranteed pod is started and that happens much less frequently.

Reply:
The current logic implementation involves (when there is no need to change the affinity):

a. getting the PID of target processes by reading `/var/run/openvswitch/ovsdb-server.pid` and `/var/run/openvswitch/ovs-vswitchd.pid`. We can improve it, but the current implementation guarantees it works with OVS daemon restarts, and it works if multiple instances of OVS are running on the node (I guess that's the reason).

b. multiple syscalls to get affinities of current and target processes.

c. syscall to get the current PID. This can be improved.

It's not very simple to determine the footprint of the 1s ticker against 5s or 10s.

Found [this interesting blog post](https://gms.tf/on-the-costs-of-syscalls.html) about syscalls cost

Reply:
@orgcandman any idea if this would be too much for OVS?

Reply:
I can't see why it could be - since the gets just read the pids - I'm not sure what else it will cause to pull from OVS.  I don't think it should be too bad.